### ISSUE TITLE
File upload button

### ISSUE THREAD

**Issue by scarroll32** [crowdAI/crowdai#119]
I was wondering if we wanted to have something a bit cooler for the file upload button? (to replace the 'choose file' button).

<img width="528" alt="screen shot 2017-05-26 at 15 35 37" src="https://cloud.githubusercontent.com/assets/11340230/26496764/14cede1c-4229-11e7-9a51-f3f2184aef42.png">

**Comment by scarroll32** [comment]
I think we should lose the 'replace' link, as you can just select a new picture. The delete is OK.
<img width="494" alt="screen shot 2017-05-26 at 16 20 08" src="https://cloud.githubusercontent.com/assets/11340230/26498426/51a9e646-422f-11e7-9615-1a99bcbe59e6.png">

Other points:
- I think max image size should be 1MB, although it can be resized on the server
- Is the cropping to make an image square? It might be nice to have a dialog if we are going to support this feature.

**Comment by jsnrynlds** [comment]
@seanfcarroll Yes, we should definitely lose the 'Replace' button when no file's been uploaded but I'd keep it when one has (as opposed to simply pressing the image).

**Comment by jsnrynlds** [comment]
@seanfcarroll I came across [WTF, forms](http://wtfforms.com/) yesterday and think the file browser button could work nicely here if I restyle it. Can you see any potential issues with this?

**Comment by scarroll32** [comment]
@jsnrynlds it looks fine, would it be possible to extract the relevant CSS and js only, so we don't have a dependency on the project?

**Comment by jsnrynlds** [comment]
@seanfcarroll So just post the HTML/CSS up here? There's no JS but the HTML would need changing to:
`<label class="file">
  <input type="file" id="file" aria-label="File browser example">
  <span class="file-custom"></span>
</label>`

**Comment by scarroll32** [comment]
Sure just post it here

**Comment by jsnrynlds** [comment]
@seanfcarroll Cool. This needs to work without Compass variables (e.g., $color-accent), right?

**Comment by jsnrynlds** [comment]
@seanfcarroll If so, here's the CSS (HTML above):
```
.file {
	position: relative;
	display: inline-block;
	cursor: pointer;
	height: 36px;
}

.file input {
	min-width: 14rem;
	margin: 0;
	filter: alpha(opacity=0);
	opacity: 0;
}

.file-custom {
	position: absolute;
	top: 0;
	right: 0;
	left: 0;
	z-index: 5;
	height: 36px;
	padding: 0 10px;
	background: rgba(0,0,0,0.03);
	border: 1px solid rgba(0,0,0,0.20);
	color: rgba(0,0,0,0.87);
	font-weight: 400;
	line-height: 36px;
	border-radius: 2px;
	box-shadow: 0 1px 1px 0 rgba(0,0,0,0.08);
}

.file-custom:after {
	content: "Choose file...";
}

.file-custom:before {
	position: absolute;
	top: -.075rem;
	right: -.075rem;
	bottom: -.075rem;
	z-index: 6;
	display: block;
	content: "Browse";
	height: 36px;
	padding: 0 20px;
	border: 1px solid rgba(0,0,0,0.20);
	background: #fff;
	color: rgba(0,0,0,0.80);
	font-size: 13px;
	font-weight: 700;
	letter-spacing: 0.5px;
	line-height: 36px;
	text-align: center;
	text-transform: uppercase;
	border-radius: 2px;
	box-shadow: 0 1px 1px 0 rgba(0,0,0,0.08);
}
```

**Comment by scarroll32** [comment]
Hi Jason

I think it would be better to work as with the rest of the CSS, so you can definitely use the compass vars 

If easier push to the crowdAI templates repo

**Comment by jsnrynlds** [comment]
@seanfcarroll No problem, here you go.

```
.file {
	position: relative;
	display: inline-block;
	cursor: pointer;
	height: 36px;
	margin: 0 0 20px;
}

.file input {
	min-width: 14rem;
	margin: 0;
	filter: alpha(opacity=0);
	opacity: 0;
}

.file-custom {
	position: absolute;
	top: 0;
	right: 0;
	left: 0;
	z-index: 5;
	height: 36px;
	padding: 0 10px;
	background: $color-grey-03;
	border: 1px solid $color-grey-20;
	color: $color-grey-87;
	font-weight: 400;
	line-height: 36px;
	@include border-radius(2px);
}

.file-custom:after {
	content: "Choose file...";
}

.file-custom:before {
	position: absolute;
	top: -.075rem;
	right: -.075rem;
	bottom: -.075rem;
	z-index: 6;
	display: block;
	content: "Browse";
	height: 36px;
	padding: 0 20px;
	border: 1px solid $color-grey-20;
	background: #fff;
	color: $color-grey-87;
	font-size: 13px;
	font-weight: 700;
	letter-spacing: 0.5px;
	line-height: 36px;
	text-align: center;
	text-transform: uppercase;
	@include border-radius(0 2px 2px 0);
}
```

**Comment by scarroll32** [comment]
@jsnrynlds Thanks for this, I have tried to implement it. However, there is an issue.

I am using a Rails form helper which is coded like this:

```
<%= f.file_field :image_file, class: 'file-custom' %>
```

This generates a name and id based on the underlying database table and column name, in this case it is like this:
```
<input class="file-custom" type="file" name="participant[image_file]" id="participant_image_file" data-validate="true">
```

The id would be different on another page.

I've tried to match the html required for the CSS to work, but it is not working. 

Is it possible to modify the styling slightly so it works with the Rails form helpers? We can discuss over the phone and perhaps work through it together?
